lib/spack/spack/package.py

```diff
@@ -605,11 +605,11 @@ def possible_dependencies(
         """
         deptype = spack.dependency.canonical_deptype(deptype)
 
-        if visited is None:
-            visited = {cls.name: set()}
+        visited = {} if visited is None else visited
+        missing = {} if missing is None else missing
+        virtuals = set() if virtuals is None else virtuals
```

Comment:
This is a bug: `virtuals` is referenced before being assigned

Reply:
fixed.